superset/views/core.py

```diff
@@ -842,6 +842,13 @@ def request_access(self):
                 .one()
             )
             datasources.add(datasource)
+
+        has_access = True
+        for datasource in datasources:
+            has_access &= (datasource and self.datasource_access(datasource))
```

Comment:
I find the `all` and `any` functions much more readable than the `&=`, something like:
`has_access = all((self.datasource_access(o) for o in datasources))`

Reply:
done